action.yml

```diff
@@ -0,0 +1,199 @@
+        -X GET "https://api.github.com/repos/$GITHUB_REPOSITORY" \
+        -H "Accept: application/vnd.github+json" \
+        -H "Authorization: Bearer ${{ github.token }}" \
+        -H "X-GitHub-Api-Version: 2022-11-28" \
+        )
+        if [[ $RESP_CODE != "200" ]]; then
+          echo "Unable to read list of tags - HTTP response code was $RESP_CODE"
+          exit 1
+        fi
+
+        # Gets all tags from GitHub API
+        # https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28
+        RESP_CODE=$(curl -w %{http_code} -s -o __tags.json \
+        -X GET "https://api.github.com/repos/$GITHUB_REPOSITORY/tags?per_page=100" \
+        -H "Accept: application/vnd.github+json" \
+        -H "Authorization: Bearer ${{ github.token }}" \
+        -H "X-GitHub-Api-Version: 2022-11-28" \
+        )
+        if [[ $RESP_CODE != "200" ]]; then
+          echo "Unable to read list of tags - HTTP response code was $RESP_CODE"
+          exit 1
+        fi
+
+        # Gets all branches from GitHub API
+        # https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#list-branches
+        RESP_CODE=$(curl -w %{http_code} -s -o __branches.json \
+        -X GET "https://api.github.com/repos/$GITHUB_REPOSITORY/branches?per_page=100" \
+        -H "Accept: application/vnd.github+json" \
+        -H "Authorization: Bearer ${{ github.token }}" \
+        -H "X-GitHub-Api-Version: 2022-11-28" \
+        )
+        if [[ $RESP_CODE != "200" ]]; then
+          echo "Unable to read list of tags - HTTP response code was $RESP_CODE"
+          exit 1
+        fi
+
+        DEFAULT_BRANCH=$(jq -r .default_branch __base.json)
+        echo "DEFAULT_BRANCH is $DEFAULT_BRANCH"
+        rm __base.json
+
+        if [[ $DEFAULT_BRANCH != $GITHUB_REF_NAME ]]; then
+          echo "Current branch $GITHUB_REF_NAME is not the same as default branch $DEFAULT_BRANCH"
+          exit 1
+        fi
+
+        # Download composer.json for use in branches.php
+        curl -s -o __composer.json https://raw.githubusercontent.com/$GITHUB_REPOSITORY/$DEFAULT_BRANCH/composer.json
+
+        BRANCHES=$(MINIMUM_CMS_MAJOR=$MINIMUM_CMS_MAJOR DEFAULT_BRANCH=$DEFAULT_BRANCH php ${{ github.action_path }}/branches.php)
+        echo "BRANCHES is $BRANCHES"
+        if [[ $BRANCHES =~ "^FAILURE \- (.+)$" ]]; then
+          MESSAGE=${BASH_REMATCH[1]}
+          echo "Exception in branches.php - $MESSAGE"
+          exit 1
+        fi
+        if [[ $BRANCHES == "" ]]; then
+          echo "No branches to merge-up"
+          exit 0
+        fi
+        echo "branches=$BRANCHES" >> $GITHUB_OUTPUT
+        rm __tags.json
+        rm __branches.json
+        rm __composer.json
+
+        # Check to see if there is anything to merge-up
+        # Technically we could skip these API calls and just do the checking using only git, though
+        # doing this way not checkout the entire git history of the repo if possible
```

Comment:
> though doing this way not checkout the entire git history of the repo if possible

Can you please reword this part? I'm not sure what this means - and it seems pretty important given the downside we're accepting as a result.

Reply:
Updated